test/unit/test_secrets_decorator.py

```diff
@@ -0,0 +1,151 @@
+import os
```

Comment:
are these run?

Reply:
absolutely. Internally we also trigger some unit tests so was wondering if you had a similar functionality as well. It's fine if it is later. Agree that unit tests are useful for dev at least.

Reply:
We do have some OB-side unit test triggers (manually), but only for the azure / gcp storage stuff (unit tests that actually exercise real cloud APIs).

I would advocate for running OSS unit tests like these automatically (add another GH actions workflow for it...).  Won't be part of this PR though.